### PR TITLE
Update to the Architecture Section

### PR DESCRIPTION
**Pull request by hannestschofenig**

### CODE REVIEW

**Review by mingpeiwk — APPROVED** [review]

File: draft-ietf-teep-architecture.md [hunk 1]
```diff
@@ -236,7 +236,14 @@ The following terms are used:
   - Trusted Application (TA): An application component that runs in a TEE.
 
   - Trusted Application (TA) Developer: An entity that wishes to provide functionality
-    on devices that requires the use of one or more Trusted Applications.
+    on devices that requires the use of one or more Trusted Applications. The TA 
+    developer signs the TA binary (or more precisely the manifest 
+    associated with the TA binary) or uses another entity on his or her behalf to get
```
Comment (dthaler) [inline draft-ietf-teep-architecture.md:241]: I disagree that the signer necessarily is "used" by the developer.   In other cases, a binary can be countersigned or re-signed by someone the TEEP Agent or TEE trusts.

File: draft-ietf-teep-architecture.md [hunk 2]
```diff
@@ -236,7 +236,14 @@ The following terms are used:
...
   - Trusted Application (TA) Developer: An entity that wishes to provide functionality
-    on devices that requires the use of one or more Trusted Applications.
+    on devices that requires the use of one or more Trusted Applications. The TA 
+    developer signs the TA binary (or more precisely the manifest 
+    associated with the TA binary) or uses another entity on his or her behalf to get 
+    the TA binary signed. (A TA binary may also be encrypted by the developer or 
+    by some third party service.) For editorial reasons, we assume that the TA 
+    developer signs the TA binary ignoring the distinction between the binary and the 
+    manifest and by simplifying the case where the TA developer outsources signing 
+    and encryption to a third party entity or service.
```
Comment (dthaler) [inline draft-ietf-teep-architecture.md:246]: I disagree that this is the approach we should take in the document.

File: draft-ietf-teep-architecture.md [hunk 3]
```diff
@@ -566,10 +576,10 @@ the device can vary. The variations depend on whether the Untrusted Application
 together or are provided separately, and this has implications to the management of
 the TAs in a TEE. In addition to the Untrusted Application and TA(s), the TA(s) and/or TEE may require
 some additional data to personalize the TA to the TA developer or the device or a user.
-This personalization data is dependent on the TEE, the TA, and the TA developer; an example of
-personalization data might be a secret symmetric key used by the TA to communicate with the TA developer. Implementations must support encryption of
+This personalization data may dependent on the type of TEE, a particular TEE instance, the TA, the TA developer and even the user of the device; an example of
```
Comment (dthaler) [inline draft-ietf-teep-architecture.md:579]: "may dependent" is bad grammar

File: draft-ietf-teep-architecture.md [hunk 4]
```diff
@@ -638,9 +650,9 @@ in a device authenticates a TAM. The
 provisioning of Trust Anchors to a device may be different from
 one use case to the other. A Device Administrator may want to
 have the capability to control what TAs are allowed.
-A device manufacturer enables verification of the TAM providers and TA binary signers; 
+A device manufacturer enables verification by one or more TAMs and by TA developers;
```
Comment (dthaler) [inline draft-ietf-teep-architecture.md:653]: I think the change from "TA binary signers" to "TA developers" should not be done ("TA signers" would be fine).

File: draft-ietf-teep-architecture.md [hunk 5]
```diff
@@ -661,9 +673,9 @@ and TEE for TAM trust verification and TA signer verification.
 ~~~~
 {: #experience title="Developer Experience"}
 
-Note that {{experience}} shows the TA developer as a TA signer.
-The TA signer is either the same as the TA developer, or is a related
-entity trusted to sign the developer's TAs.
+Note that {{experience}} shows the view from a TA developer point of view. 
+The TA developer signs the TA or is a related
+entity trusted to sign the developer-created TAs.
```
Comment (dthaler) [inline draft-ietf-teep-architecture.md:678]: This sentence doesnt make any sense now.

File: draft-ietf-teep-architecture.md [hunk 6]
```diff
@@ -804,7 +816,7 @@ that is chained to a certificate that the TAM trusts.
 
 This architecture uses a PKI, although self-signed certificates are
 also permitted.  Trust Anchors exist on the devices to
-enable the TEE to authenticate TAMs and TA signers, and TAMs use Trust Anchors to
+enable the TEE to authenticate TAMs and TA developer, and TAMs use Trust Anchors to
```
Comment (dthaler) [inline draft-ietf-teep-architecture.md:819]: bad grammar, and I disagree with the new text.

**Pull request merged**